spec/factories/campaign.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :campaign do
+    active { true }
+    valid { true }
+    end_date { DateTime.now }
```

Comment:
Faker has a way to generate a random date in the future: `Faker::Time.forward(days: 30)   # any time up to 30 days into the future`